src/api/animals.controller.ts

```diff
@@ -1,6 +1,16 @@
 import type { Request, Response } from 'express'
-import { animalsData } from '../data/animals-data'
+import { animalsData, type Animal } from '../data/animals-data'
 
 export const getAnimals = (_req: Request, res: Response): void => {
   res.status(200).json(animalsData)
 }
+
+const filterAnimals = (animals: Animal[], filter: string): Animal[] => {
+  return animals.filter(animal => animal.name.toLowerCase().includes(filter.toLowerCase()))
+}
+
+export const getAnimalsByFilter = (req: Request, res: Response): void => {
+  const filter = req.params.filter
+  const filteredAnimals = filterAnimals(animalsData, filter)
+  res.status(200).json(filteredAnimals)
```

Comment:
I pushed changes in commit 84ed3957e48a650b3501ec9715300c4e059ba102

> The `getAnimalsByFilter` endpoint is correctly implemented. However, it's important to validate and sanitize the `filter` parameter extracted from `req.params` to prevent potential security vulnerabilities, such as injection attacks.